src/Microsoft.AspNetCore.Server.Kestrel/Http/MessageBody.cs

```diff
@@ -353,12 +462,34 @@ private static bool TakeChunkedLine(SocketInput baton, ref int chunkSizeOut)
                 }
             }
 
+            private static void ThrowInvalidFormat()
+            {
+                throw new InvalidOperationException("Bad Request");
+            }
+
+            private static void ThrowChunkedRequestIncomplete()
+            {
+                throw new InvalidOperationException("Chunked request incomplete");
+            }
+
+            private static void ThrowChunkedExtensionsNotSupported()
+            {
+                throw new NotImplementedException("Chunked-extensions not supported");
+            }
+
+            private static void ThrowTrailingHeadersNotSupported()
+            {
+                throw new NotImplementedException("Trailing headers not supported");
```

Comment:
Introduce some new errors? `HttpErrorException(numeric, message, deets)` wondering how easy it is to currently distinguish things and their various levels of meh -> serious